src/modules/user-settings/components/preferences-tab.tsx

```diff
@@ -5,51 +5,57 @@ import { useI18nData } from 'src/hooks/use-i18n-data';
 import { EditorPicker } from 'src/modules/user-settings/components/editor-picker';
 import { LanguagePicker } from 'src/modules/user-settings/components/language-picker';
 import { TerminalPicker } from 'src/modules/user-settings/components/terminal-picker';
-import { useEditorData } from 'src/modules/user-settings/hooks/use-editor-data';
-import { useTerminalData } from 'src/modules/user-settings/hooks/use-terminal-data';
+import { SupportedEditor } from 'src/modules/user-settings/lib/editor';
+import {
+	useGetUserEditorQuery,
+	useGetUserTerminalQuery,
+	useSaveUserEditorMutation,
+	useSaveUserTerminalMutation,
+} from 'src/stores/installed-apps-api';
 
 export const PreferencesTab = ( { onClose }: { onClose: () => void } ) => {
 	const { __ } = useI18n();
 	const { locale: savedLocale, setLocale: setSavedLocale } = useI18nData();
 	const [ locale, setLocale ] = useState( savedLocale );
 
-	const { editor, handleEditorChange, saveEditorPreference, resetEditor, hasEditorChanges } =
-		useEditorData();
+	const { data: editor } = useGetUserEditorQuery();
+	const { data: terminal = 'terminal' } = useGetUserTerminalQuery();
+	const [ saveEditor ] = useSaveUserEditorMutation();
+	const [ saveTerminal ] = useSaveUserTerminalMutation();
 
-	const {
-		terminal,
-		handleTerminalChange,
-		saveTerminalPreference,
-		resetTerminal,
-		hasTerminalChanges,
-		availableTerminals,
-	} = useTerminalData();
+	const [ currentEditor, setCurrentEditor ] = useState< SupportedEditor | undefined >(
+		editor ?? undefined
+	);
+	const [ currentTerminal, setCurrentTerminal ] = useState( terminal );
 
 	const savePreferences = async () => {
 		setSavedLocale( locale );
-		await saveEditorPreference();
-		await saveTerminalPreference();
+		if ( currentEditor ) {
+			await saveEditor( currentEditor );
+		}
+		await saveTerminal( currentTerminal );
 		onClose();
 	};
 
 	const cancelChanges = () => {
 		setLocale( savedLocale );
-		resetEditor();
-		resetTerminal();
+		setCurrentEditor( editor ?? undefined );
+		setCurrentTerminal( terminal );
 		onClose();
 	};
 
-	const hasChanges = locale !== savedLocale || hasEditorChanges || hasTerminalChanges;
+	const hasChanges =
+		locale !== savedLocale || currentEditor !== editor || currentTerminal !== terminal;
 
 	return (
 		<>
 			<LanguagePicker value={ locale } onChange={ setLocale } />
-			<EditorPicker value={ editor } onChange={ handleEditorChange } />
-			<TerminalPicker
-				value={ terminal }
-				onChange={ handleTerminalChange }
-				availableTerminals={ availableTerminals }
+			<EditorPicker
+				value={ currentEditor }
+				onChange={ setCurrentEditor }
+				disabled={ editor === undefined }
```

Comment:
```suggestion
				disabled={ isEditorLoading }
```
This line had me confused to begin with because of the "fallback logic" in `getUserEditor`. I realize now that what we're saying here is "disable this control while the editors are loading". I think it would be clearer if we used the `isLoading` prop returned by `useGetUserEditorQuery` for that purpose.

Reply:
We had a quick call with @gavande1 yesterday and I realized that I forgot to move over the fallback logic from `getUserEditor`. My intention with these changes was to account for a case where there's no selected editors, and no supported editors are installed on the system either.

For this case, we have a Figma design based on which we should disable the dropdown, and show a CTA to install VSCode. @gavande1 worked on updating the UI with these in [STU-436 User Settings: Add CTA for installing VSCode if none of supported editor is installed](https://linear.app/a8c/issue/STU-436/user-settings-add-cta-for-installing-vscode-if-none-of-supported), but in this PR I `getUserEditor` with the fallback logic and added a `undefined` case.